presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkConfig.java

```diff
@@ -477,4 +492,27 @@ public PrestoSparkConfig setAdaptiveJoinSideSwitchingEnabled(boolean adaptiveJoi
         this.adaptiveJoinSideSwitchingEnabled = adaptiveJoinSideSwitchingEnabled;
         return this;
     }
+
+    /**
+     * Parses the catalog properties from the input string in format catalog.key=value and returns a nested map of properties.
+     */
+    public static Map<String, Map<String, String>> parseCatalogProperties(String input)
```

Comment:
Is there any existing utility method that does this.
cc : @rschlussel

Reply:
https://github.com/prestodb/presto/blob/master/presto-spark-launcher/src/main/java/com/facebook/presto/spark/launcher/LauncherUtils.java#L60-L84 is maybe the closest